client/src/components/workspaces/utils.ts

```diff
@@ -79,10 +86,42 @@ export async function workspaceShareClick(workspace: WorkspaceInfo, informationM
   await modal.onWillDismiss();
 }
 
+export async function workspaceFavoriteClick(
```

Comment:
`ToggleFavorite` or something like that, this function can be called without a click.